### ISSUE TITLE
`mix xref trace` wrong line #

### ISSUE THREAD

**Issue by marcandre** [elixir-lang/elixir#11260]
### Environment

Tested with elixir @ master (30647679e3c)

### Current behavior

```
$ MIX_ENV=test mix xref trace lib/bobby_web/endpoint.ex --label compile
lib/bobby_web/endpoint.ex:1: call BobbyWeb.Plugs.Maintenance.__info__/1 (compile)
lib/bobby_web/endpoint.ex:1: call BobbyWeb.Plugs.Maintenance.__info__/1 (compile)
lib/bobby_web/endpoint.ex:1: call BobbyWeb.Router.__info__/1 (compile)
lib/bobby_web/endpoint.ex:1: call BobbyWeb.Router.__info__/1 (compile)
lib/bobby_web/endpoint.ex:84: alias BobbyWeb.CacheBodyReader (compile)
```

### Expected behavior

Have a more useful line # showing...

Our `endpoint.ex` starts with:
```elixir
defmodule BobbyWeb.Endpoint do
  use Phoenix.Endpoint, otp_app: :bobby
  # ...
```

**Comment by josevalim** [comment]
This is not an Elixir bug. Phoenix collects the modules and generates functions calls in a @before_compile callback. All code in such callback without line gets the module line (cause we don’t have anything else). We may need to fix this in Plug and Phoenix. I will leave this open so I don’t forget to explore this.

**Comment by marcandre** [comment]
Makes sense. I'll close this, as this seems resolved in Phoenix 1.16-rc.0
